scripts/process_schemas.py

```diff
@@ -201,44 +207,60 @@ def write(self, outputFile):
             superclass = "SearchRequest"
         elif self.isSearchResponse():
             superclass = "SearchResponse"
-        string = "\n\nclass {0}({1}):".format(self.schema.name, superclass)
+        self._writeNewline(outputFile, 2)
+        string = "class {0}({1}):".format(self.schema.name, superclass)
         print(string, file=outputFile)
         doc = self.schema.doc
         if doc is None:
             doc = "No documentation"
-        string = '    """\n{0}\n    """'.format(doc)
-        print(string, file=outputFile)
+        self._writeWithIndent('"""', outputFile)
+        self._writeWrappedWithIndent(doc, outputFile)
+        self._writeWithIndent('"""', outputFile)
         if isinstance(self.schema, avro.schema.RecordSchema):
-            string = '    _schemaSource = """\n{0}"""'.format(
+            string = '_schemaSource = """\n{0}"""'.format(
                 self.formatSchema())
-            print(string, file=outputFile)
-            string = '    schema = avro.schema.parse(_schemaSource)'
-            print(string, file=outputFile)
-            string = '    requiredFields = {0}'.format(
-                self.formatRequiredFields())
-            print(string, file=outputFile)
+            self._writeWithIndent(string, outputFile)
+            string = 'schema = avro.schema.parse(_schemaSource)'
+            self._writeWithIndent(string, outputFile)
+            self.writeRequiredFields(outputFile)
             if self.isSearchResponse():
-                string = '    _valueListName = "{0}"'.format(
+                string = '_valueListName = "{0}"'.format(
                     self.getValueListName())
-                print(string, file=outputFile)
-            print(file=outputFile)
+                self._writeWithIndent(string, outputFile)
+            self._writeNewline(outputFile)
             self.writeEmbeddedTypesClassMethods(outputFile)
             self.writeConstructor(outputFile)
         elif isinstance(self.schema, avro.schema.EnumSchema):
             # TODO make a proper Python enum here using the Python 3.4 enum?
             for symbol in self.schema.symbols:
-                string = '    {0} = "{0}"'.format(symbol, symbol)
-                print(string, file=outputFile)
+                string = '{0} = "{0}"'.format(symbol, symbol)
+                self._writeWithIndent(string, outputFile)
+
+    def _writeWithIndent(self, string_, outputFile, indentLevel=1):
```

Comment:
I tried making that refactor, but here's the problem: the `SchemaClass` instance does not know if it should open that file for writing or appending.  That is, it doesn't know if it is the first `SchemaClass` instance created, and why should it have to know anything about the order of other instances' creation?  Aha, so you say, why not just pass in the file stream?  Well, `SchemaClass` instantiation and writing are done in two different places, and the file is only opened when the later is called.  So if we wanted to eliminate all of the `outputFile` argument passing (which I agree is annoying), we would probably need to change the interface.  I'm not super opposed to doing that, but my main intent with this change was to just make the file more readable so nobody has to count white space to debug the output, etc.